src/gamestate/commands.cpp

```diff
@@ -3141,12 +3141,82 @@ std::vector<dcon::province_id> can_move_army(sys::state& state, dcon::nation_id
+			// handled 1.0 - Enemy territory that does not have troops at the time of withdrawal (regulated in mp)
+			// handled 1.1 - Allied/your territory that is being occupied in a war
+			// extern - To ships (including when a naval battle is in progress)
+			// Not being able to withdraw:
+			// handled 1.2 - When all provinces adjacent to a battle are surrounded by enemies
+			// handled 1.3 - No more than 1 province, to avoid multiprovince
+			// handled 1.4 - Where there are enemy troops (your territory / ally or enemy, it doesn't matter)
+			bool b_10 = false; // Also handles 1-1, occupied territory of allies or yours
+			// 1.0/1.1 - Enemy territory
+			if(military::are_at_war(state, state.world.province_get_nation_from_province_control(dest), source)) {
+				auto units = state.world.province_get_army_location_as_location(dest);
+				b_10 = true;  // Enemy territory with no enemy units -- can retreat
+				for(const auto unit : units) {
+					if(unit.get_army().get_controller_from_army_control() == source)
+						continue;
+					if(unit.get_army().get_controller_from_army_rebel_control()
+					|| military::are_at_war(state, unit.get_army().get_controller_from_army_control(), source)) {
+						b_10 = false;  // Enemy territory with enemy units -- CAN'T retreat
+						break;
+					}
+				}
+			}
+			// 1.2 - Sorrounding/encirclement of land units
+			bool b_12 = false;
+			for(const auto adj : state.world.province_get_province_adjacency(dest)) {
+				auto other = adj.get_connected_provinces(adj.get_connected_provinces(0) == dest ? 1 : 0);
+				if(other.id.index() < state.province_definitions.first_sea_province.index()) {
+					auto units = state.world.province_get_army_location_as_location(dest);
+					bool has_enemy_units = false;
+					for(const auto unit : units) {
+						if(unit.get_army().get_controller_from_army_rebel_control()
+						|| military::are_at_war(state, unit.get_army().get_controller_from_army_control(), source)) {
+							has_enemy_units = true;
+							break;
+						}
+					}
+					if(!has_enemy_units) { //Not a full encirclement -- can retreat
+						b_12 = true;
+						break;
+					}
+				}
+			}
+			// 1.3 - Not more than 1 province
+			bool b_13 = false;
+			for(const auto adj : state.world.province_get_province_adjacency(dest)) {
+				auto other = adj.get_connected_provinces(adj.get_connected_provinces(0) == dest ? 1 : 0);
+				if(last_province == other) {
+					b_13 = true; //Is adjacent to destination, hence a single province retreat!?
+					break;
+				}
+			}
+
+			if(state.world.army_get_black_flag(a)) {
+				return province::make_unowned_land_path(state, last_province, dest);
+			} else if(province::has_access_to_province(state, source, dest) && b_12 && b_13) {
+				return province::make_land_path(state, last_province, dest, source, a);
+			} else if(b_10) {
+				return province::make_unowned_land_path(state, last_province, dest);
+			} else {
+				return std::vector<dcon::province_id>{};
+			}
 		} else {
```

Comment:
The way you have re ordered the conditionals, I do not believe this is correct. I believe that you would get here if the army is in battle and you are not in multiplayer, which would allow partial retreats outside of multiplayer

Reply:
``		if(state.network_mode != sys::network_mode_type::single_player
		&& state.world.army_get_battle_from_army_battle_participation(a)) {``

Reply:
ok fixes